### PR TITLE
Add activation functions doc reference

### PR DESCRIPTION
**Pull request by NeroBlackstone**
issue #517

### CODE REVIEW

File: docs/src/.vitepress/config.mts [hunk 1]
```diff
@@ -192,6 +192,10 @@ export default defineConfig({
                             { text: 'Interoperability', link: '/api/Lux/interop' },
                             { text: 'DistributedUtils', link: '/api/Lux/distributed_utils' }]
                     },
+                    {
+                        text: 'Activation Functions', collapsed: false, items: [
+                            { text: 'Activation Functions', link: 'https://fluxml.ai/NNlib.jl/dev/reference/#Activation-Functions' }]
+                    },
```
Comment (avik-pal) [inline docs/src/.vitepress/config.mts:198]: Put it under `Building Blocks` as:

```
{ text: 'NNlib', link: 'https://fluxml.ai/NNlib.jl/dev/' },
{ text: 'Activation Functions', link: 'https://fluxml.ai/NNlib.jl/dev/reference/#Activation-Functions' }
```

**Review by avik-pal — CHANGES_REQUESTED** [review]

**Comment by NeroBlackstone** [comment]
Add docs for [Parallel](https://lux.csail.mit.edu/stable/api/Lux/layers#Lux.Parallel)

**Comment by NeroBlackstone** [comment]
Add [FlattenLayer](https://lux.csail.mit.edu/stable/api/Lux/layers#Lux.FlattenLayer) Example

**Comment by NeroBlackstone** [comment]
Add [ReshapeLayer](https://lux.csail.mit.edu/stable/api/Lux/layers#Lux.ReshapeLayer) docs

**Comment by avik-pal** [comment]
Doctests are failing, you would want to compare sizes or something that doesn't vary due to randomness. Else there is a way to avoid checking exact values via Documenter.jl but I don't remember the exact syntax for that

**Comment by NeroBlackstone** [comment]
The error output is so long that I can't locate the problem. I removed the random numbers in `jldoctest`, or directly removed `jldoctest` to `julia`, but I manually tested all the added example codes and there was no problem.

File: src/layers/containers.jl [hunk 2]
```diff
@@ -122,6 +122,26 @@ with `connection`.
...
+
+julia> using Random;
+       rng = Random.seed!(123);
+       ps, st = Lux.setup(rng, model);
+       x1 = randn(rng,Float32,2);
+       x2 = randn(rng,Float32,2);
+
+julia> y, st_new = model((x1,x2),ps,st)
+((Float32[2.0603316], Float32[0.7091646]), (layer_1 = NamedTuple(), layer_2 = NamedTuple()))
+```
```
Comment (avik-pal) [inline src/layers/containers.jl:144]:
```suggestion
```jldoctest
julia> model = Parallel(nothing,Dense(2,1),Dense(2,1))
Parallel(
    layer_1 = Dense(2 => 1),            # 3 parameters
    layer_2 = Dense(2 => 1),            # 3 parameters
)         # Total: 6 parameters,
          #        plus 0 states.

julia> using Random;
       rng = Random.seed!(123);
       ps, st = Lux.setup(rng, model);
       x1 = randn(rng,Float32,2);
       x2 = randn(rng,Float32,2);

julia> size.(first(model((x1, x2),ps,st)))
((1,), (1,))
```
```

**Comment by avik-pal** [comment]
Thanks for these updates, I rebased it in https://github.com/LuxDL/Lux.jl/pull/702/. Will merge it there.

**Pull request closed without merging**